apps/web/pages/event-types/[type]/index.tsx

```diff
@@ -176,48 +177,59 @@ const EventTypePage = (props: EventTypeSetupProps) => {
-      hidden: eventType.hidden,
-      periodDates: {
-        startDate: periodDates.startDate,
-        endDate: periodDates.endDate,
-      },
-      periodType: eventType.periodType,
-      periodCountCalendarDays: eventType.periodCountCalendarDays ? "1" : "0",
-      schedulingType: eventType.schedulingType,
-      minimumBookingNotice: eventType.minimumBookingNotice,
-      metadata,
-      hosts: !!eventType.hosts?.length
-        ? eventType.hosts.filter((host) => !host.isFixed)
-        : eventType.users
-            .filter(() => eventType.schedulingType === SchedulingType.ROUND_ROBIN)
-            .map((user) => ({ userId: user.id })),
-      hostsFixed: !!eventType.hosts?.length
-        ? eventType.hosts.filter((host) => host.isFixed)
-        : eventType.users
-            .filter(() => eventType.schedulingType === SchedulingType.COLLECTIVE)
-            .map((user) => ({ userId: user.id })),
+  const defaultValues = {
+    title: eventType.title,
+    locations: eventType.locations || [],
+    recurringEvent: eventType.recurringEvent || null,
+    description: eventType.description ?? undefined,
+    schedule: eventType.schedule || undefined,
+    bookingLimits: eventType.bookingLimits || undefined,
+    length: eventType.length,
+    hidden: eventType.hidden,
+    periodDates: {
+      startDate: periodDates.startDate,
+      endDate: periodDates.endDate,
     },
+    bookingFields: eventType.bookingFields,
+    periodType: eventType.periodType,
+    periodCountCalendarDays: eventType.periodCountCalendarDays ? "1" : "0",
+    schedulingType: eventType.schedulingType,
+    minimumBookingNotice: eventType.minimumBookingNotice,
+    metadata,
+    hosts: !!eventType.hosts?.length
+      ? eventType.hosts.filter((host) => !host.isFixed)
+      : eventType.users
+          .filter(() => eventType.schedulingType === SchedulingType.ROUND_ROBIN)
+          .map((user) => ({ userId: user.id })),
+    hostsFixed: !!eventType.hosts?.length
+      ? eventType.hosts.filter((host) => host.isFixed)
+      : eventType.users
+          .filter(() => eventType.schedulingType === SchedulingType.COLLECTIVE)
+          .map((user) => ({ userId: user.id })),
+  } as const;
+
+  const formMethods = useForm<FormValues>({
+    defaultValues,
     resolver: zodResolver(
       z
         .object({
           // Length if string, is converted to a number or it can be a number
           // Make it optional because it's not submitted from all tabs of the page
           length: z.union([z.string().transform((val) => +val), z.number()]).optional(),
+          bookingFields: eventTypeBookingFields,
         })
         // TODO: Add schema for other fields later.
         .passthrough()
     ),
   });
 
+  useEffect(() => {
+    if (!formMethods.formState.isDirty) {
+      //TODO: What's the best way to sync the form with backend
+      formMethods.setValue("bookingFields", defaultValues.bookingFields);
+    }
+  }, [defaultValues]);
+
```

Comment:
Don't render the form until defaultValues is not undefined.

```suggestion
```

[Try the React Container approach](https://reactpatterns.js.org/docs/container-component)